R/backend-snowflake.R

```diff
@@ -18,13 +18,193 @@ NULL
+              0, "Sunday"
+            )
+          )
+        } else if (label && abbr) {
+          sql_expr(DAYNAME(!!x))
+        } else {
+          abort("Unrecognized arguments to `wday`")
+        }
+      },
+      yday = function(x) sql_expr(EXTRACT("dayofyear", !!x)),
+      week = function(x) {
+        sql_expr(FLOOR((EXTRACT("dayofyear", !!x) - 1L) / 7L) + 1L)
+      },
+      isoweek = function(x) sql_expr(EXTRACT("weekiso", !!x)),
+      # For consistency with postgres, right-pad month names for uniform size
+      month = function(x, label = FALSE, abbr = TRUE) {
+        if (!label) {
+          sql_expr(EXTRACT("month", !!x))
+        } else {
+          if (abbr) {
+            sql_expr(MONTHNAME(!!x))
+          } else {
+            sql_expr(
+              DECODE(
+                EXTRACT("month", !!x),
+                1, "January  ",
+                2, "February ",
+                3, "March    ",
+                4, "April    ",
+                5, "May      ",
+                6, "June     ",
+                7, "July     ",
+                8, "August   ",
+                9, "September",
+                10, "October  ",
+                11, "November ",
+                12, "December "
+              )
+            )
+          }
+        }
+      },
+      quarter = function(x, with_year = FALSE, fiscal_start = 1) {
+        if (fiscal_start != 1) {
+          abort("`fiscal_start` is not supported in Snowflake translation. Must be 1.")
+        }
+
+        if (with_year) {
+          sql_expr((EXTRACT("year", !!x) || "." || EXTRACT("quarter", !!x)))
+        } else {
+          sql_expr(EXTRACT("quarter", !!x))
+        }
+      },
+      isoyear = function(x) {
+        sql_expr(EXTRACT("year", !!x))
+      },
+      seconds = function(x) {
+        build_sql("INTERVAL '", x, " second'")
+      },
+      minutes = function(x) {
+        build_sql("INTERVAL '", x, " minute'")
+      },
+      hours = function(x) {
+        build_sql("INTERVAL '", x, " hour'")
+      },
+      days = function(x) {
+        build_sql("INTERVAL '", x, " day'")
+      },
+      weeks = function(x) {
+        build_sql("INTERVAL '", x, " week'")
+      },
+      months = function(x) {
+        build_sql("INTERVAL '", x, " month'")
+      },
+      years = function(x) {
+        build_sql("INTERVAL '", x, " year'")
+      },
+      # https://docs.snowflake.com/en/sql-reference/functions/date_trunc.html
+      floor_date = function(x, unit = "seconds") {
+        unit <- arg_match(
+          unit, c("second", "minute", "hour", "day", "week", "month", "quarter", "year")
+        )
+        sql_expr(DATE_TRUNC(!!unit, !!x))
+      }
+    ),
+    sql_translator(
+      .parent = base_agg,
+      cor = sql_aggregate_2("CORR"),
+      cov = sql_aggregate_2("COVAR_SAMP"),
+      all = sql_aggregate("BOOLAND_AGG", "all"),
+      any = sql_aggregate("BOOLOR_AGG", "any"),
+      exactly_one = sql_aggregate("BOOLXOR_AGG", "exactly_one"),
```

Comment:
There's no R function `exactly_one` so I wouldn't include this in the translation.

Reply:
Sounds good, will remove!